src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -736,9 +736,12 @@ private BoundStatement BindDeclarationStatementParts(LocalDeclarationStatementSy
         internal MethodSymbol TryFindDisposePatternMethod(BoundExpression expr, SyntaxNode syntaxNode, bool hasAwait, DiagnosticBag diagnostics)
         {
             Debug.Assert(!(expr is null));
-            if (expr.Type is null)
+                                
+            // For dispose on nullable value types, we always perform lookup on the underlying type, as we'll never actually call in the null case
+            // this emulates the same behavior as x?.Dispose(); 
+            if (expr.Type.IsNullableType())
             {
-                return null;
+                expr = CreateConversion(expr, expr.Type.GetNullableUnderlyingType(), diagnostics);
```

Comment:
>CreateConversion [](start = 23, length = 16)

This surprised me. I think we generally only use conversions when they are specified in the language. But this conversion is not be part of the spec.
You're really just trying to get a BoundNode that is an expression and that has the right type. You could just create a placeholder node of the right type to achieve that goal (ie. create a new bound node similar to `BoundAwaitableValuePlaceholder`).